platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -188,7 +187,7 @@ public AbstractTestTask() {
         reports.getJunitXml().getRequired().set(true);
         reports.getHtml().getRequired().set(true);
 
-        filter = instantiator.newInstance(DefaultTestFilter.class);
+        filter = getProject().getObjects().newInstance(DefaultTestFilter.class);
```

Comment:
🤔 Why are we using a different instantiator now?

Reply:
When using the original instantiator we get an error:
```
A problem occurred evaluating root project 'ido1x'.
> Could not create task ':test'.
   > Could not create task of type 'Test'.
      > Could not create an instance of type org.gradle.api.internal.tasks.testing.filter.DefaultTestFilter.
         > No service of type ObjectFactory available in default services.
```

Reply:
I am not sure why we used instantiator which is internal class anyway. Also I am not sure why in this scope instantiator doesn't have ObjectFactory, but that is a different problem